src/libpriv/rpmostree-core.c

```diff
@@ -1998,6 +2331,23 @@ checksum_version (GVariant *checksum)
   return g_strdup (ret);
 }
 
+static guint
+pkg_hash (gconstpointer v)
+{
+  DnfPackage *p = (DnfPackage*)v;
+  return g_str_hash (dnf_package_get_package_id (p));
+}
+
+static gboolean
+pkg_equal (gconstpointer v1,
+           gconstpointer v2)
+{
+  DnfPackage *p1 = (DnfPackage*)v1;
+  DnfPackage *p2 = (DnfPackage*)v2;
+  return g_str_equal (dnf_package_get_package_id (p1),
+                      dnf_package_get_package_id (p2));
```

Comment:
So, the reason why all this was needed in the first place was because `dnf_package_get_chksum()` didn't work on our dummy RPMs (which is what's used in `rpmostree_dnf_add_checksum_goal()`). I decided to dig deeper as to why that was, which lead to https://github.com/rpm-software-management/libdnf/pull/273. Once this is merged, we can back out a lot of the logic there.